### ISSUE TITLE
TEXT-encoding of 'aa9' and 'aa`' identical

### ISSUE THREAD

**Issue by adrianschlatter** [datalog/datamatrix-svg#4]
Encoding the messages 'aa9' and 'aa`' to TEXT gives identical results for both which can't be correct.

```
toText('aa`', [ /* TEXT */ 239, 31, 0, 0, 32, 9, 29, 47, 1, 33, 57, 9, 44, 64, 1, 43, 90, 2, 64, 95, 1, 69, 122, 9, 83, 127, 2, 96, 255, 1, 0])
```
=> [239, 89, 190, 254]

```
toText('aa9', [ /* TEXT */ 239, 31, 0, 0, 32, 9, 29, 47, 1, 33, 57, 9, 44, 64, 1, 43, 90, 2, 64, 95, 1, 69, 122, 9, 83, 127, 2, 96, 255, 1, 0])
```
=> [239, 89, 190, 254]

**Comment by datalog** [comment]
Thank you fFirst of all thank you for your feedback.

It is true that this function may return the same result for different chunks of data. Please, keep in mind that this is not standalone function, this is just a part of a cascade. Every next step will revaluate data again in order to choose optimum encoding method... This also helps to avoid mentioned collision.
or your

**Comment by datalog** [comment]
First of all thank you for your feedback.

It is true that this function may return the same result for different chunks of data. Please, keep in mind that this is not standalone function, this is just a part of a cascade. Every next step will revaluate data again in order to choose optimum encoding method... This also helps to avoid mentioned collision.

**Comment by adrianschlatter** [comment]
I agree with you. However, it is apparently not enough to *assure* correctness. I tried a bit with your nice [datamatrix generator tool](https://datalog.github.io/demo/datamatrix-svg) and found a collision (i.e. same datamatrix) for these strings:

* "aaaaaaaaaaa9"
* "aaaaaaaaaaa`"
